### DISCUSSION TITLE
Absl compilation error in Visual Studio 2022 - 17.7.3

### DISCUSSION THREAD

**Discussion by Diddlik** [google/or-tools#3922] [Build and configuration issues] [answered]
**What version of OR-Tools and what language are you using?**
Version: 9.6 / 9.7
Language: C++

**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)**
GLOP

**What operating system (Linux, Windows, ...) and version?**
Windows 11

**What did you do?**
    Updated Visual Studio 2022 from version 17.5.3 to 17.7.3.
    Tried compiling the project which uses OR-Tools library.
    See compilation errors.

If I open the same solution file on the same computer but with VS 17.5.3 then it compiles without errors.
Tried both C++ 20 and C++17

**What did you expect to see**
Successful compilation without any errors, similar to the behavior in Visual Studio 2022 version 17.5.3.

**What did you see instead?**
Received multiple errors, for instance:
Error C2574 '(__cdecl *absl::lts_20230125::time_internal::cctz::detail::civil_time<absl::lts_20230125::time_internal::cctz::detail::day_tag>::civil_time(void))(void)': cannot be declared static (compiling source file Source\ML_Solver_Args.cpp) GLOP_Lib I:\Coding\DGO\GLOP_Lib\include\absl\time\internal\cctz\include\cctz\civil_time_detail.h 408  

![grafik](https://github.com/google/or-tools/assets/6010713/9a7a8bc7-cbf5-4441-badb-bb8af888861f)

Make sure you include information that can help us debug (full error message, model Proto).

**Anything else we should know about your project / environment**

**Comment by evakili** [comment, marked as answer]
I have the same problem in VS 17.7.3. Removing parenthesis around `min` and `max` in lines 404 and 408 will fix the issue. Anyway, I have no idea why this error occurred.

**Reply by Diddlik** [reply]
thanks, now it compiles.

**Reply by evakili** [reply]
You should be aware that removing those parentheses may cause conflicts with `min` or `max` macros defined in other headers (e.g. Windows.h). Those parentheses are there to manage those conflicts. This means that removing parentheses is only a workaround, not a solution.

**Reply by tuket** [reply]
So what is the proper solution?
